src/k8s/pkg/component/hook.go

```diff
@@ -5,16 +5,17 @@ import (
 	"net"
 	"strings"
 
+	"github.com/canonical/k8s/pkg/snap"
 	"github.com/canonical/k8s/pkg/utils"
 )
 
-type valuesHook func() (map[string]any, error)
+type valuesHook func(snap.Snap) (map[string]any, error)
 
 var valuesHooks = map[string]valuesHook{
 	"network": networkValues,
 }
 
-func networkValues() (map[string]any, error) {
+func networkValues(sn snap.Snap) (map[string]any, error) {
```

Comment:
`sn` does not read very well. `snap` is better here, also other places. it's fine if we shadow the `snap` package name.

Reply:
or perhaps `s` if context is well-known

Reply:
There is a lot of potential for collisions. 
s -> micrcluster state
Simply shadowing the snap package also is not always possible as we have `snap.MySnapFunction` and `snap := snapFromContext`

Where required, I renamed the package to `snapPkg` and shadowed the `snap`name otherwise.